src/App.js

```diff
@@ -4,26 +4,41 @@ import './App.css';
 import chatMessages from './data/messages.json';
 import ChatLog from './components/ChatLog';
 
-const App = () => {
-  const [likes, setLikes] = useState(0);
-  const incrementLikes = () => {
-    setLikes(likes + 1);
+const countLikes = (messages) => {
+  let count = 0;
+  for (const message of messages) {
+    if (message.liked) {
+      count += 1;
+    }
   }
-  const decrementLikes = () => {
-    setLikes(likes - 1);
+  return count;
+};
```

Comment:
👍 Making a helper function can really help with our code clarity. Needing to count the number of something in a list is one of the things the built-in, general-purpose `reduce` function can help with, but we certainly aren't required to use it. But for comparison, here's an approach of how we might implement `countLikes` using `reduce`
```js
const countLikes = (messages) => {
  // adding a bool to a number treats true as 1 and 0 as false
  return messages.reduce((totalLikes, message) => (totalLikes + message.liked), 0);
};
```

While we could use the `reduce` approach inline (where you are calling `countLikes`), I still think there's value in putting it into a well-named helper to make it more clear what's going on.

Between
```js
  const likes = countLikes(messages);
```
and 
```js
  const likes = messages.reduce((totalLikes, message) => (totalLikes + message.liked), 0);
```
I know which _I_ find more clear, and it's not the one with `reduce`.